### PR TITLE
ENGDESK-32328: Improve RTP endpoint and Park + Eavesdrop behavior

### PR DESCRIPTION
**Pull request by dev-ryanc**
- Make local ip and port parameter as optional
 - Handle properly mode parameter
 - Added new api function to allow changing remote address and port
 - Fix compare_val funtion
 - Add configuration for default local ip in rtp endpoint
 - Check `send_silence_when_idle` at runtime during Park state
 - Eavesdrop in whisper mode will automatically apply `send_silence_when_idle` if needed during Park state

### CODE REVIEW

File: src/mod/endpoints/mod_sofia/rtp.c [hunk 1]
```diff
@@ -129,20 +130,127 @@ switch_io_routines_t crtp_io_routines = {
...
+					} else {
+						switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_ERROR, "Unknown IP for interface %s for rtp-ip\n", val+10);
+					}
+				} else {
+					ip = strcasecmp(val, "auto") ? val : mod_sofia_globals.guess_ip;
+				}
+
+				if (!zstr(ip)) {
+					crtp.default_local_ip = switch_core_strdup(crtp.pool, ip);
+                    switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_INFO, "RTP Endpoint will be using default ip: %s\n", ip);
```
Comment (damirn) [inline src/mod/endpoints/mod_sofia/rtp.c:217]: here also

File: src/mod/endpoints/mod_sofia/rtp.c [hunk 2]
```diff
@@ -129,20 +130,127 @@ switch_io_routines_t crtp_io_routines = {
...
+				}
+			}
+		}
+	}
+
+done:
+    
+    if (zstr(crtp.default_local_ip)) {
```
Comment (damirn) [inline src/mod/endpoints/mod_sofia/rtp.c:225]: here also?

File: src/mod/endpoints/mod_sofia/rtp.c [hunk 3]
```diff
@@ -479,7 +609,7 @@ static switch_status_t channel_receive_event(switch_core_session_t *session, swi
         if (compare_var(event, channel, kCODEC) ||
             compare_var(event, channel, kPTIME) ||
             compare_var(event, channel, kPT) ||
-	    compare_var(event, channel, kRATE)) {
+	        compare_var(event, channel, kRATE)) {
```
Comment (damirn) [inline src/mod/endpoints/mod_sofia/rtp.c:612]: this whole file seems to be a combination of tabs and spaces :) but i'd say that the intention was to use spaces for indent

File: src/mod/endpoints/mod_sofia/rtp.c [hunk 4]
```diff
@@ -129,20 +130,127 @@ switch_io_routines_t crtp_io_routines = {
...
+    stream->write_function(stream, "-USAGE: %s\n", CRTP_MEDIA_MODIFY_SYNTAX);
 
-void crtp_init(switch_loadable_module_interface_t *module_interface)
+done:
+
+	switch_safe_free(mycmd);
+	return SWITCH_STATUS_SUCCESS;
+}
+
+
```
Comment (damirn) [inline src/mod/endpoints/mod_sofia/rtp.c:176]: perhaps to remove an extra line?

**Review by damirn — APPROVED** [review]
looks nice; minor indent fixes needed

File: src/mod/endpoints/mod_sofia/rtp.c [hunk 5]
```diff
@@ -129,20 +130,127 @@ switch_io_routines_t crtp_io_routines = {
...
 
+#define CRTP_MEDIA_MODIFY_SYNTAX "<uuid> <remote ip> <remote port>"
+SWITCH_STANDARD_API(crtp_media_modify_function)
+{
+	switch_core_session_t *psession = NULL;
+    char *mycmd = NULL, *argv[3] = { 0 };
+	int argc = 0;
+
+	if (!zstr(cmd) && (mycmd = strdup(cmd))) {
+		argc = switch_separate_string(mycmd, ' ', argv, (sizeof(argv) / sizeof(argv[0])));
```
Comment (izzytelnyx) [inline src/mod/endpoints/mod_sofia/rtp.c:141]: We might need to check zstr(argv[0]) before line 141... I see that we've been checking it in line 142.

**Review by izzytelnyx — APPROVED** [review]

File: src/mod/endpoints/mod_sofia/rtp.c (hunk 5 above)
Comment (dev-ryanc) [reply src/mod/endpoints/mod_sofia/rtp.c:141]: argv supposted to be empty before line 41. The method `switch_separate_string` will perform tokenization and store the output to that array. Once this is executed, thats the time I validated the content of the array argv

File: src/mod/endpoints/mod_sofia/rtp.c (hunk 3 above)
Comment (dev-ryanc) [reply src/mod/endpoints/mod_sofia/rtp.c]: Yes the whole file is half spaces, half tab. Fixing the spaces will just confuse the reviewer. Better create a separate PR or do this after merging the latest FS version

**Review by izzytelnyx — APPROVED** [review]

**Pull request merged**